projects/com.oracle.truffle.llvm.pipe.native/src/capture.cpp

```diff
@@ -0,0 +1,57 @@
+#include <jni.h>
+#include "native.h"
+
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <iostream>
+using namespace std;
+#include <string>
+
+#define BUF_SIZE 1024
+
+enum PIPES { R, W };
+static int myPipe[2];
+static int oldStdOut;
+static std::string captured;
+
+JNIEXPORT void JNICALL Java_com_oracle_truffle_llvm_pipe_CaptureOutput_startCapturing(JNIEnv *env, jclass c) {
+	pipe(myPipe);
+	oldStdOut = dup(fileno(stdout));
+	fflush(stdout);
+	dup2(myPipe[W], fileno(stdout));
+}
+
+JNIEXPORT void JNICALL Java_com_oracle_truffle_llvm_pipe_CaptureOutput_stopCapturing(JNIEnv * env, jclass c) {
+ 	fflush(stdout);
+    dup2(oldStdOut, fileno(stdout));
+    captured.clear();
+
+    if (myPipe[W] > 0)
+        close(myPipe[W]);
+    std::string buf;
+    buf.resize(BUF_SIZE);
+    
+    int bytesRead = 0;
+    do {
+    	bytesRead = read(myPipe[R], &(*buf.begin()), BUF_SIZE);
+    	captured += buf;     	
+    } while(bytesRead == BUF_SIZE);
+
+    if (bytesRead > 0) {
+        buf.resize(bytesRead);
+        captured += buf;
+    }
+    if (myPipe[R] > 0)
+        close(myPipe[R]);
+}
+
+
+JNIEXPORT jstring JNICALL Java_com_oracle_truffle_llvm_pipe_CaptureOutput_getCapture(JNIEnv * env, jclass c) {
+	std::string::size_type idx = captured.find_last_not_of("\r\n");
```

Comment:
Why `\r\n`? Is this not just applicable for Windows as a line break, while Linux uses `\n`? Can we have a named constant for that?